src/cljs/main/broadfcui/page/workspace/notebooks/notebooks_table.cljs

```diff
@@ -0,0 +1,443 @@
+
+(react/defc NotebooksContainer
+  {:refresh
+   (fn [{:keys [this]}]
+     (this :-get-pet-token)
+     (this :-get-clusters-list)
+     (this :-schedule-cookie-refresh))
+
+   :render
+   (fn [{:keys [props state this]}]
+     (let [{:keys [server-response cluster-map show-create-dialog?]} @state
+           {:keys [notebooks server-error]} server-response]
+       [:div {:display "inline-flex"}
+        (when show-create-dialog?
+          [NotebookCreator (assoc props :dismiss #(swap! state dissoc :show-create-dialog?)
+                                        :pet-token (:pet-token @state)
+                                        :notebooks notebooks
+                                        :refresh-notebooks #(this :-refresh-notebooks))])
+        [:div {} [:span {:data-test-id "notebooks-title" :style {:fontSize "125%" :fontWeight 500 :paddingBottom 10 :marginLeft 10}} "Notebooks"]]
+        [:div {:style {:margin 10 :fontSize "88%"}}
+         "Create a Jupyter notebook and launch an interactive analysis environment based on Spark and Hail.
+         See online documentation " [:a {:href (config/user-notebooks-guide-url) :target "_blank"} "here" icons/external-link-icon]]
+        [comps/ErrorViewer {:data-test-id "notebooks-error" :error server-error}]
+        (if notebooks
+          [NotebooksTable
+           (assoc props
+             :toolbar-items [flex/spring [buttons/Button {:data-test-id "create-modal-button"
+                                                          :text "Create Notebook..." :style {:marginRight 1}
+                                                          :onClick #(swap! state assoc :show-create-dialog? true)}]
+                             ; TODO: upload notebook not yet implemented
+                             [:div {} [buttons/Button {:data-test-id "upload-modal-button"
+                                                       :text "Upload Notebook..." :style {:marginRight 1}}]]]
+             :choose-cluster #(this :-assoc-cluster-with-notebook %1 %2)
+             :stop-cluster #(this :-stop-cluster %)
+             :start-cluster #(this :-start-cluster %)
+             :reload-after-delete #(this :-get-clusters-list %)
+             :cluster-map cluster-map
+             :notebooks notebooks
+             :pet-token (:pet-token @state)
+             :refresh-notebooks #(this :-refresh-notebooks))]
+          [:div {:style {:textAlign "center"}} (spinner "Loading notebooks...")])]))
+
+   :component-did-mount
+   (fn [{:keys [this]}]
+     (this :-get-pet-token)
+     (this :-get-clusters-list)
+     (this :-schedule-cookie-refresh)
+     (.addEventListener js/window "message" (react/method this :-notebook-extension-listener)))
+
+   :component-will-unmount
+   (fn [{:keys [this locals]}]
+     (swap! locals assoc :dead? true)
+     (.removeEventListener js/window "message" (react/method this :-notebook-extension-listener)))
+
+   ; Communicates with the Leo notebook extension.
```

Comment:
A lot of this code is copied from here: https://github.com/broadinstitute/firecloud-ui/blob/develop/src/cljs/main/broadfcui/page/workspace/notebooks/notebooks.cljs#L458